src/GazeboUnderwater.cpp

```diff
@@ -28,74 +26,110 @@ namespace gazebo_underwater
         if(sdf->HasElement(_parameter_name.c_str()))
         {
             var = sdf->Get< T >(_parameter_name.c_str());
-            gzmsg << "GazeboUnderwater: " + _parameter_name + ": (" << var << ") " + dimension  <<std::endl;
-        }        
-        else
-        {
-            gzmsg << "GazeboUnderwater: " + _parameter_name + ": using default (" << default_value << ") " + dimension << std::endl;
+            gzmsg << "GazeboUnderwater: " + _parameter_name + ": (" << var << ") "
+                    + dimension  << std::endl;
+        }else{
+            gzmsg << "GazeboUnderwater: " + _parameter_name + ": using default ("
+                    << default_value << ") " + dimension << std::endl;
         }
         return var;
     }
 
-
     void GazeboUnderwater::loadParameters(void)
     {
         if(sdf->HasElement("model_name"))
         {
             model = world->GetModel( sdf->Get<std::string>("model_name") );  
-            gzmsg << "GazeboUnderwater: found model:" << model->GetName() << std::endl;
+            gzmsg << "GazeboUnderwater: found model: " << model->GetName() << std::endl;
+        }else{
+            gzmsg << "GazeboUnderwater: model_name to apply buoyancy not defined !" << std::endl;
+        }
+
+        link = model->GetLink( sdf->Get<std::string>("link_name") );
+        if( link ){
+            gzmsg << "GazeboUnderwater: found link: " << link->GetName() << std::endl;
+            physics::InertialPtr modelInertia = link->GetInertial();
+            gzmsg << "GazeboUnderwater: link mass: " << modelInertia->GetMass() << std::endl;
         }else{
-            gzmsg << "GazeboUnderwater: model not found. Quit simulation... " << std::endl;
+            gzmsg << "GazeboUnderwater: no model link matches link_name !" << std::endl;
         }
 
         size = getParameter<math::Vector3>("size","meters",
                 math::Vector3(1, 1, 1));
         centerOfBuoyancy = getParameter<math::Vector3>("center_of_buoyancy","meters",
-                math::Vector3(0, 0, 0.1));
+                math::Vector3(0, 0, 0.15));
         fluidVelocity = getParameter<math::Vector3>("fluid_velocity","m/s",
                 math::Vector3(0,0,0));
         viscousDamping = getParameter<math::Vector3>("viscous_damping","dimensionless",
                 math::Vector3(20,30,50));
         waterLevel = getParameter<double>("water_level","meters", 2.0);
         densityOfFluid = getParameter<double>("fluid_density","kg/m3", 1000);
+        // buoyancy must be the buoyancy when the model is completely submersed
+        buoyancy = getParameter<double>("buoyancy","N",0);
     }
 
     void GazeboUnderwater::updateBegin(common::UpdateInfo const& info)
     {
-        physics::LinkPtr link = model->GetLink( sdf->Get<std::string>("link_name") ); 
-        applyBuoyancy(link);
-        applyViscousFriction(link);
+        if(link)
```

Comment:
This plugin will not find the link if one misspell the link name, i.e. if the link name inside the plugin tag doesn't match the link model name. (I actually need to test two things: if model_name is defined and if model_name name a model loaded in gazebo - todo).
I don't know if I should quit the simulation, because we can add a feature to apply buoyancy to multiple links/models. So, if we are testing the simulation we can comment a link/model inside the world file and still run the simulation. What is your opinion ?